tests/tape/test_qnode.py

```diff
@@ -323,12 +323,52 @@ def func(x):
             qml.RX(x, wires=0)
             return qml.expval(qml.PauliZ(0))
 
-        qn = qml.QNode(func, dev)
+        qn = qml.QNode(func, dev, diff_method="finite-diff")
 
         assert (
             qn.__repr__()
-            == "<QNode: wires=1, device='default.qubit.autograd', interface='autograd', diff_method='best'>"
+            == "<QNode: wires=1, device='default.qubit', interface='autograd', diff_method='finite-diff'>"
         )
+        assert qn.diff_method_change == False
+
+    def test_qnode_best_diff_method(self, monkeypatch):
+        """Test that selected "best" diff_method is available after QNode initialization."""
+        dev = qml.device("default.qubit", wires=1)
+
+        def func(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        qn = qml.QNode(func, dev)
+
+        assert qn.diff_method == "backprop"

```

Comment:
I believe you could just assert that `qn.diff_method == qn.diff_options["method"]`, instead of manually entering the expected diff method. This would rather focus on testing that the `diff_method` attribute is correctly changed.

Reply:
`qn.diff_method == qn.diff_options["method"]`  can not be done.
As things are currently set,
diff_method will only always be one of the options that the user can manually set during initialization (e.g. "finite-diff", "parameter-shift", etc.).
diff_options["method"] can be other things depending on the inner workings like "numeric" and "analytic", which the user can not set manually.
The current use case of the diff_method for the issue is to show which method has been selected from the ones available to the user.
I think diff_method and diff_options["method"] can be updated to show the same thing always, but that needs to be its own issue.

Reply:
There are four tests for four `diff_method`s. This will merge the first two, but the other two will remain separate. I think having four individual tests keeps it clean and simple.